File: src/main/kotlin/stift/io/home/adapters/LLMOrchestrator.kt
package stift.io.home.adapters

import org.springframework.stereotype.Service
import stift.io.home.adapters.agents.storage.StorageAgentImpl

// TODO: Doesn't orchestrate much as of now - we only have one agent 🤷‍♂️
// TODO: But think: What are the reasons to break functionality up into multiple agents?
@Service
class LLMOrchestrator(
    private val storageAgent: StorageAgentImpl,
) {
    fun call(request: String): String {
        return this.storageAgent.call(
            request = request
        )
    }
}

File: python_app/llm_orchestrator.py
from python_app.agents.storage_agent import PythonStorageAgent

class PythonLLMOrchestrator:
    def __init__(self, storage_agent: PythonStorageAgent):
        self.storage_agent = storage_agent

    def call(self, request: str) -> str:
        return self.storage_agent.chat(request)
